includes/REST_API/Products_Controller.php

```diff
@@ -0,0 +1,489 @@
+		}
+
+		if ( function_exists( 'wc_get_products' ) ) {
+			return $this->get_items_woocommerce( $request );
+		}
+
+		return rest_ensure_response( $results );
+	}
+
+	/**
+	 * Retrieves all Shopify products.
+	 *
+	 * @SuppressWarnings(PHPMD.ExcessiveMethodLength)
+	 *
+	 * @since 1.20.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	protected function get_items_shopify( $request ) {
+		/**
+		 * Search term.
+		 *
+		 * @var string $search_term
+		 */
+		$search_term = $request['search'];
+
+		/**
+		 * Filters the shopify products data TTL value.
+		 *
+		 * @since 1.20.0
+		 *
+		 * @param int $time Time to live (in seconds). Default is 1 day.
+		 * @param WP_REST_Request $request Full details about the request.
+		 */
+		$cache_ttl = apply_filters( 'web_stories_shopify_data_cache_ttl', DAY_IN_SECONDS, $request );
+		$cache_key = ! empty( $request['search'] ) ? 'web_stories_shopify_data_' . md5( $search_term ) : 'web_stories_shopify_data_default';
+
+		$data = get_transient( $cache_key );
+
+		if ( \is_string( $data ) && ! empty( $data ) ) {
+			/**
+			 * Decoded cached products data.
+			 *
+			 * @var array|null $products
+			 */
+			$products = json_decode( $data, true );
+
+			if ( $products ) {
+				return rest_ensure_response( $products );
+			}
+		}
+
+		$results = [];
+
+		/**
+		 * Shopify host.
+		 *
+		 * @var string $host
+		 */
+		$host = $this->settings->get_setting( Settings::SETTING_NAME_SHOPIFY_HOST );
+
+		/**
+		 * Shopify access token.
+		 *
+		 * @var string $access_token
+		 */
+		$access_token = $this->settings->get_setting( Settings::SETTING_NAME_SHOPIFY_ACCESS_TOKEN );
+
+		// TODO(#11154): Maybe move to a class constant.
+		$api_version = '2022-01';
+
+		// TODO(#11154): Proper sanitization.
+		$url = sprintf(
+			'https://%1$s/api/%2$s/graphql.json',
+			$host,
+			$api_version
+		);
+
+		$search_string = empty( $search_term ) ? '*' : '*' . $search_term . '*';
+
+		// TODO(#11154): Support different sortKeys.
+		$query = <<<QUERY
+{
+  products(first: 100, sortKey: CREATED_AT, query: "available_for_sale:true AND title:$search_string") {
+    edges {
+      node {
+        id
+        title
+        vendor
+        description
+        priceRange {
+          minVariantPrice {
+            amount
+            currencyCode
+          }
+        }
+        onlineStoreUrl
+        images(first: 10) {
+          edges {
+            node {
+              url
+              altText
+            }
+          }
+        }
+      }
+    }
+  }
+}
+QUERY;
+
+		$response = wp_remote_post(
+			$url,
+			[
+				'headers' => [
+					'Content-Type'                      => 'application/graphql',
+					'X-Shopify-Storefront-Access-Token' => $access_token,
+				],
+				'body'    => $query,
+			]
+		);
+
+		if ( WP_Http::OK !== wp_remote_retrieve_response_code( $response ) ) {
```

Comment:
Not sure about dumping some errors as we don't typically do that elsewhere (yet), but definitely would need more robust error handling. Probably as part of #11154.

Reply:
I think we should try and work out what the error was here and show different errors. 
Example, 
1. auth error. 
2. Rate limit error
3. 404 / missing item error. 
4. 500 error. 

If we are proxying there api, I think it would be extremely useful for debug, if we don't just swallow that error.